typescript-rewrite/samlang-core/src/interpreter/__tests__/expression-interpreter.test.ts

```diff
@@ -0,0 +1,796 @@
+import {
+  MUL,
+  DIV,
+  MOD,
+  PLUS,
+  MINUS,
+  LT,
+  LE,
+  GT,
+  GE,
+  EQ,
+  NE,
+  AND,
+  CONCAT,
+  OR,
+} from '../../ast/common/binary-operators';
+import Position from '../../ast/common/position';
+import Range from '../../ast/common/range';
+import {
+  boolType,
+  intType,
+  stringType,
+  unitType,
+  FunctionType,
+  identifierType,
+  TupleType,
+} from '../../ast/common/types';
+import {
+  SamlangExpression,
+  EXPRESSION_INT,
+  EXPRESSION_STRING,
+  EXPRESSION_TRUE,
+  EXPRESSION_THIS,
+  EXPRESSION_VARIABLE,
+  EXPRESSION_CLASS_MEMBER,
+  EXPRESSION_TUPLE_CONSTRUCTOR,
+  EXPRESSION_OBJECT_CONSTRUCTOR,
+  EXPRESSION_VARIANT_CONSTRUCTOR,
+  EXPRESSION_FIELD_ACCESS,
+  EXPRESSION_METHOD_ACCESS,
+  EXPRESSION_UNARY,
+  EXPRESSION_BUILTIN_FUNCTION_CALL,
+  EXPRESSION_FUNCTION_CALL,
+  EXPRESSION_BINARY,
+  EXPRESSION_IF_ELSE,
+  EXPRESSION_FALSE,
+  EXPRESSION_MATCH,
+  VariantPatternToExpression,
+  EXPRESSION_LAMBDA,
+  EXPRESSION_PANIC,
+  EXPRESSION_STATEMENT_BLOCK,
+  StatementBlock,
+  SamlangValStatement,
+} from '../../ast/lang/samlang-expressions';
+import {
+  TuplePattern,
+  ObjectPattern,
+  ObjectPatternDestucturedName,
+  VariablePattern,
+  WildCardPattern,
+} from '../../ast/lang/samlang-pattern';
+import ExpressionInterpreter from '../expression-interpreter';
+import { ClassValue, EMPTY, InterpretationContext } from '../interpretation-context';
+import { isSameValue, Value, FunctionValue } from '../value';
+
+const interpreter = new ExpressionInterpreter();
+
+const exampleRange: Range = new Range(new Position(1, 2), new Position(3, 4));
+const intLiteralExpression: SamlangExpression = EXPRESSION_INT(exampleRange, BigInt(5));
+const stringLiteralExpression: SamlangExpression = EXPRESSION_STRING(exampleRange, 'value');
+const boolLiteralExpression: SamlangExpression = EXPRESSION_TRUE(exampleRange);
+const classMemberFunction: Value = {
+  type: 'functionValue',
+  arguments: [],
+  body: EXPRESSION_INT(exampleRange, BigInt(5)),
+  context: EMPTY,
+};
+const objectConstructorExpressionNonEmpty = EXPRESSION_OBJECT_CONSTRUCTOR({
+  range: exampleRange,
+  type: intType,
+  fieldDeclarations: [
+    { range: exampleRange, type: intType, name: 'test', expression: intLiteralExpression },
+  ],
+});
+const functionType: FunctionType = {
+  type: 'FunctionType',
+  argumentTypes: [unitType],
+  returnType: stringType,
+};
+const functionExpression = EXPRESSION_LAMBDA({
+  range: exampleRange,
+  type: functionType,
+  parameters: [],
+  captured: {},
+  body: stringLiteralExpression,
+});
+
+it('literal expressions evaluate correctly', () => {
+  expect(
+    isSameValue(interpreter.eval(intLiteralExpression), { type: 'int', value: 5 })
```

Comment:
Now I know why you want to implement `isSameValue`. In fact you don't need it. Jest's `expect(...).toEqual(...)` performs a deep structural equality check.

Reply:
🤦‍♀️

Reply:
removed `isSameValue` and related deep equals functions for `interpretation-context` as well